lib/child_process.js

```diff
@@ -60,22 +60,29 @@ function stdioStringToArray(option) {
 }
 
 exports.fork = function fork(modulePath /* , args, options */) {
-
   // Get options and args arguments.
   var execArgv;
-  var options = {};
-  var args = [];
-  var pos = 1;
-  if (pos < arguments.length && Array.isArray(arguments[pos])) {
-    args = arguments[pos++];
-  }
 
-  if (pos < arguments.length && arguments[pos] != null) {
-    if (typeof arguments[pos] !== 'object') {
-      throw new ERR_INVALID_ARG_VALUE(`arguments[${pos}]`, arguments[pos]);
-    }
+  var args = arguments[1];
+  var options = arguments[2];
+
+  if (Array.isArray(args)) {
+    args = args.slice(0);
+  } else if (options !== undefined &&
+    (args === undefined ||
+      args === null ||
+      typeof args !== 'object')
```

Comment:
If `args` is not an object and it is not `undefined` or `null`, it should actually throw an error. Just ignoring the argument does not help anyone.

Reply:
Adding a 
``` js
throw new ERR_INVALID_ARG_VALUE('arg', 'object', options);
```
for the code you mentioned breaks this line in test - 

https://github.com/nodejs/node/blob/468ab4519e1b92473acefb22801497a1af6aebae/test/parallel/test-child-process-spawn-typeerror.js#L180

Should we update this test and add ```assert.thorws``` for this case.

Reply:
No, the test cases would continue to work but the code has to be rewritten further.